docs/proposals/20-ha-control-planes.md

```diff
@@ -365,6 +366,21 @@ With most of the complexity being handled by [Etcd-Druid](https://github.com/gar
 
 The groundwork for this was already done by [gardener/gardener#5741](https://github.com/gardener/gardener/pull/5741).
 
+### Other critical components having single replica
+
+Following shoot control plane components are currently setup with a single replica and are planned to run with a minimum of 2 replicas:
+
+* Cluster Autoscaler (if enabled)
+* Cloud Controller Manager (CCM)
+* Kube Controller Manager (KCM)
+* Kube Scheduler
+* Machine Controller Manager (MCM)
+
+> NOTE: MCM and CCM are components deployed by provider extensions. HA specific configuration should be configured there.
```

Comment:
We could also configure this via another webhook in gardener-resource-manager/gardener-seed-admission-controller instead adding the same check over and over in all extensions :)

Reply:
Changed. Also added some corrections w.r.t node drain.

Reply:
Do we really need a webhook for this? Code duplication can be avoided in many ways. But I don't really see a need for a new webhook for a configuration option for 5 Deployments. It is yet another failure point.

Reply:
We also proposed this webhook for https://github.com/gardener/gardener/blob/master/docs/proposals/20-ha-control-planes.md#zone-pinning as well. This will be an added responsibility

Reply:
I cannot agree conceptually. So in GRM webhook you will need to hardcode the deployment names (CCM, MCM, etc). which breaks the extensibility concept. Doing so would also require the extensions to deploy CCM and MCM with leader election enabled. From Gardener point of view you cannot make assumptions if leader election for these components is enabled or not. And what if an extension needs to deploy yet another control plane component is HA mode with 2 replicas?
So I would rather prefer to do not do it via webhook and let the extensions tackle for CCM and MCM and gardenlet - for the rest (CA, KCM, kube-scheduler). It simply introduces tight coupling.